src/core/ddsi/src/q_plist.c

```diff
@@ -344,17 +378,27 @@ static void fini_generic_partial (void * __restrict dst, size_t * __restrict dst
       case XSTOP: return;
       case XO: COMPLEX (XO, ddsi_octetseq_t, ddsrt_free (x->value), (void) 0); break;
       case XS: COMPLEX (XS, char *, ddsrt_free (*x), (void) 0); break;
-      case XZ: COMPLEX (XZ, ddsi_stringseq_t, { for (uint32_t i = 0; i < x->n; i++) ddsrt_free (x->strs[i]); }, ddsrt_free (x->strs)); break;
       case XE1: case XE2: case XE3: COMPLEX (*desc, unsigned, (void) 0, (void) 0); break;
       case Xi: case Xix2: case Xix3: case Xix4: SIMPLE (Xi, int32_t); break;
       case Xu: case Xux2: case Xux3: case Xux4: case Xux5: SIMPLE (Xu, uint32_t); break;
-      case Xl: SIMPLE (Xl, int32_t); break;
       case XD: case XDx2: SIMPLE (XD, dds_duration_t); break;
       case Xo: case Xox2: SIMPLE (Xo, unsigned char); break;
       case Xb: case Xbx2: SIMPLE (Xb, unsigned char); break;
       case XbCOND: SIMPLE (XbCOND, unsigned char); break;
       case XG: SIMPLE (XG, nn_guid_t); break;
       case XK: SIMPLE (XK, nn_keyhash_t); break;
+      case XbPROP: SIMPLE (XbPROP, unsigned char); break;
+      case XQ:
+        /* non-nested, so never a need to deallocate only some of the entries and no complications
+           in locating the end of the sequence element description */
+        COMPLEX (XQ, ddsi_octetseq_t, {
+          size_t elem_off = 0;
+          for (uint32_t i = 0; i < x->length; i++) {
+            fini_generic_partial (x->value, &elem_off, desc + 1, desc_end, aliased);
```

Comment:
My mistake was thinking that the calls to deser_generic_src() takes care of that, but that's only for the alignment/padding elements within the structures, not 'between' them.  

The reason my manual test works is because the first element within the (binary)property is the string, causing a 4byte alignment automatically.

I've extended the manual test with the same kind of struct you proposed in your comment. That shows the problem clearly.

Element offset calculation revert coming up.